internal/pkg/vm/internal/vmcontext/actor_state_handle.go

```diff
@@ -77,35 +125,57 @@ func (h *actorStateHandle) UpdateRelease(obj interface{}) {
 		runtime.Abort("Storage commit error")
 	}
 
-	// make new head available to take
+	// update head
 	h.head = &newcid
+
+	// add validation
+	h.addCidValidation(obj, newcid)
```

Comment:
A list of validation fns is not the right semantics here. If a method contains two well-formed transactions in sequence then the validation fn from the first one will not pass after the second. I think we want semantics of a single expected CID which is updated with each transaction.